compiler/src/main/kotlin/dev/zacsweers/metro/compiler/fir/checkers/AssistedInjectChecker.kt

```diff
@@ -71,7 +71,7 @@ internal object AssistedInjectChecker : FirClassChecker(MppCheckerKind.Common) {
       }
     if (injectConstructor == null) {
       reporter.reportOn(
-        targetType.source,
```

Comment:
The issue is that `targetType.source` could be null if the current FIR session cannot resolve, for example Unit (missing return type).

Changing the report source to the assisted factory function fixed it, I think it's more suitable than `targetType` which might be unpredictable.

Another solution will be splitting the checks, we keep the current `targetType.source` reporting when it's presented, otherwise report the unexpected type when `targetType.source` is missing. Let me know what you prefer.

Reply:
Think this is fine to start 👍